consensus/src/round_manager.rs

```diff
@@ -526,27 +522,19 @@ impl RoundManager {
             self.round_state.current_round()
         );
 
-        let parent_block = self
-            .block_store
-            .get_block(executed_block.parent_id())
-            .expect("[RoundManager] Parent block not found after execution");
-
         let vote_proposal = VoteProposal::new(
-            AccumulatorExtensionProof::<TransactionAccumulatorHasher>::new(
-                parent_block.compute_result().frozen_subtree_roots().clone(),
-                parent_block.compute_result().num_leaves(),
-                executed_block
-                    .compute_result()
-                    .transaction_info_hashes()
-                    .clone(),
-            ),
+            executed_block.compute_result().extension_proof(),
```

Comment:
it looks executed_block can generate the vote_proposal and maybe_signed_vote_proposal directly?
```
fn vote_proposal(&self) -> VoteProposal {
}
```